solr/solr-ref-guide/src/shards-and-indexing-data-in-solrcloud.adoc

```diff
@@ -122,6 +122,8 @@ More details on how to use shard splitting is in the section on the Collection A
 
 In most cases, when running in SolrCloud mode, indexing client applications should not send explicit commit requests. Rather, you should configure auto commits with `openSearcher=false` and auto soft-commits to make recent updates visible in search requests. This ensures that auto commits occur on a regular schedule in the cluster.
 
+NOTE: Using auto soft commit or commitWithin requires the client app to embrace the realities of "eventual consistency". Solr will make documents searchable at _roughly_ the same time across NRT replicas of a collection but there are no hard guarantees. Consequently, in rare cases, it's possible for a document to show up in one search only for it not to appear in a subsequent search occurring immediately after when the second search is routed to a different replica. Also, documents added in a particular order (even in the same batch) might become searchable out of the order of submission when there is sharding.
```

Comment:
"auto soft commit"

IDK whether it's worth including hard-commit-with-opensearcher=true

Reply:
I've made some edits on your second two points.   I don't know that I'm comfortable enough with all the semantics to tackle your `hard-commit-with-opensearch=true` variant.   Please do update the branch however!   Otherwise, I'd like to commit this early next week.